go/common/crypto/signature/signers/ledger/ledger_signer.go

```diff
@@ -125,7 +132,15 @@ func (s *Signer) ContextSign(context signature.Context, message []byte) ([]byte,
 	if err != nil {
 		return nil, err
 	}
-	return s.device.SignEd25519(s.path, preparedContext, message)
+	signature, err := s.device.SignEd25519(s.path, preparedContext, message)
+	if err != nil {
+		if err.Error() == ledgerRejectTxErrorMsg {
```

Comment:
Maybe use `errors.Is` instead. Also consider using a switch statement instead of these nested ifs.

Reply:
I've removed nested ifs and used a switch statement.

I couldn't use `errors.Is()` because at the moment, ledger-go doesn't use proper Go error semantics and doesn't expose errors as variables.